renderdoc/replay/entry_points.cpp

```diff
@@ -890,6 +1203,208 @@ extern "C" RENDERDOC_API void RENDERDOC_CC RENDERDOC_StartAndroidRemoteServer(co
       "shell am start -n org.renderdoc.renderdoccmd/.Loader -e renderdoccmd remoteserver");
 }
 
+extern "C" RENDERDOC_API bool RENDERDOC_CC RENDERDOC_CheckAndroidLayer(const char *host,
+                                                                       const char *exe)
+{
+  string packageName(basename(string(exe)));
+
+  int index = 0;
+  std::string deviceID;
+  Android::extractDeviceIDAndIndex(host, index, deviceID);
+
+  // Find the path to package
+  string pkgPath = trim(adbExecCommand(deviceID, "shell pm path " + packageName).strStdout);
+  pkgPath.erase(pkgPath.begin(), pkgPath.begin() + strlen("package:"));
+  pkgPath.erase(pkgPath.end() - strlen("base.apk"), pkgPath.end());
+  pkgPath += "lib";
+
+  string layerName = "libVkLayer_GLES_RenderDoc.so";
+
+  // First, see if the application contains the layer
+  if(SearchForAndroidLayer(deviceID, pkgPath, layerName))
+    return true;
+
+  // Next, check a debug location only usable by rooted devices
+  if(SearchForAndroidLayer(deviceID, "/data/local/debug/vulkan", layerName))
+    return true;
+
+  // TODO: Add any future layer locations
+
+  RDCWARN("No RenderDoc layer for Vulkan was found");
+  return false;
+}
+
+string DetermineInstalledABI(const string &deviceID, const string &packageName)
+{
+  RDCLOG("Checking installed ABI for %s", packageName.c_str());
+  string abi;
+
+  string dump = adbExecCommand(deviceID, "shell pm dump " + packageName).strStdout;
```

Comment:
Yes, I was thinking of adding an option that disables logging for large commands like this.

I should be able to move to execCommand and not print the output to log, a quick workaround.